src/main/java/com/redhat/red/build/finder/BuildFinder.java

```diff
@@ -257,16 +258,17 @@ public BuildFinder(ClientSession session, BuildConfig config) {
                 try {
                     if (builds.containsKey(archive.getBuildId())) {
                         build = builds.get(archive.getBuildId());
-                        LOGGER.info("Build id {} already in table with {} archives", archive.getBuildId(), build.getArchives() != null ? build.getArchives().size() : 0);
                         buildInfo = build.getBuildInfo();
                         taskInfo = build.getTaskInfo();
                         taskRequest = build.getTaskRequest();
                         archiveList = build.getArchives();
                         tags = build.getTags();
                         hits++;
 
+                        LOGGER.info(Ansi.ansi().render("Found build: id: @|green {}|@ nvr: @|green {}|@ checksum: @|green {}|@ archive: @|green {}|@ @|cyan [{} / {} = {}%]|@").toString(), buildInfo.getId(), buildInfo.getNvr(), checksum, archive.getFilename(), checked, total, String.format("%.3f", (checked / (double) total) * 100));
```

Comment:
I wonder if the new instance creation/toString negates some of the benefit of lazy parameter evaluation?

At first glance the logging seems more obtuse ; I wonder if calling the formatters as parameters would be preferable?:

```LOGGER.info ("Found build  {} {} {} nvr: its  ", Ansi.ansi().fgBlue(), buildinfo.getId() , Ansi.ansi().reset());```

I haven't looked to see if we could implement a custom formatter to further simplify this.

Note also I've seen some of those escapes appear in output e.g. "[WARN] Unable to process archive/compressed file: @|red symbolic.zip!/wget/wget.1.gz|@".

Reply:
One thing is that I think making the ANSI codes parameters makes it quite a bit harder to read. And while it should not affect the lazy parameter evaluation per se, as that is left untouched, it must add time due to the fact that it pre-renders the string and then does the substitution. We'd have to benchmark it, I guess.

It's more of a case of the fact that this code is meant for *interactive* output. It is not a server that is performing millions of transactions a second to where a few more milliseconds inside the logger method would matter.

I just went the simplest route.

Reply:
I agree the time spent is probably not worth worrying about for an interactive application. If we were ever to make this a server would we need to reconsider?

However I guess we just have different views - I think "@|green {}|@" is harder to reader than "{}{}{}" :-) Although  to be fair I've not a massive fan of each in terms of readabillity. I don't know if there is any way of improving either of those two forms or what @jcoleman0redhat0com thinks.

Not sure whether we could wrap the parameter e.g. `green(buildInfo.getId())` but then to make that lazy evaluated you'd still need to create a new object and use string concatenation.....trade off between readability and object creation.

Reply:
I like this most recent idea the best out of what has been suggesed. But, is it worth creating a new class for the colored logging methods? The "@||@" syntax is just a convenience to begin with, though not the most efficient, allows for the least amount of extra code to construct a message.

Reply:
@dwalluck Agreed  "@||@" is least amount of code but its also not very readable ;-) Do you mean wrapping the above static functions inside a helper class?

Reply:
> @dwalluck Agreed "@||@" is least amount of code but its also not very readable ;-) Do you mean wrapping the above static functions inside a helper class?

Right. I believe the color functions similar to the above example should be inside a helper class, as they may be used in more than one class.

I'll try to implement something like this where the formatting can apply to the argument and we can leave the actual message portion as it was before the color changes.